sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -505,28 +442,27 @@ object IntervalUtils {
               i += 1
             case _ if '0' <= b && b <= '9' =>
               isNegative = false
-            case _ => return null
+            case '.' =>
```

Comment:
can we have a separated PR to fix bug? Now it's a little hard to review with everything mixed together.

Reply:
OK, i'd fix this bug first

Reply:
https://github.com/apache/spark/pull/26514